pinot-transport/src/main/java/com/linkedin/pinot/transport/pool/AsyncPoolResourceManagerAdapter.java

```diff
@@ -85,19 +91,30 @@ public boolean validatePut(T obj) {
 
   @Override
   public void destroy(final T obj, final boolean error, final Callback<T> callback) {
-    _executor.submit(new Runnable() {
-
-      @Override
-      public void run() {
-        LOGGER.info("Running teardown for the client connection " + obj + " Error is : " + error);
-        boolean success = _resourceManager.destroy(_key, error, obj);
-        if (success) {
-          callback.onSuccess(obj);
-        } else {
-          callback.onError(new Exception("Unable to destroy resource for key " + _key));
+    try {
+      _executor.submit(new Runnable() {
+
+        @Override
+        public void run() {
+          LOGGER.info("Running teardown for the client connection " + obj + " Error is : " + error);
+          boolean success = _resourceManager.destroy(_key, error, obj);
+          if (success) {
+            callback.onSuccess(obj);
+          } else {
+            callback.onError(new Exception("Unable to destroy resource for key " + _key));
+          }
         }
+      });
+    } catch (Exception e) {
+      // During a broker shutdown, it is possible that we get RejectedExecutionException since we are tyring to
```

Comment:
RejectedExecutionException is posisble if the executor is being shutdown. If you call _executor.shutdown() in shutdown method maybe you can move  if (_isShuttingDown) check at the top of the method.